### ISSUE TITLE
Invalid insertion when comparing identical single-element arrays

### ISSUE THREAD

**Issue by flovouin** [wokalski/Diff.swift#22]
Hi,

Thanks for the great library, it is indeed a lot faster than Dwifft. As I'm still using Swift 2.3, I'm currently on `b153507`, however the following is an issue that is still present on `master`.
When comparing two arrays containing a single, identical element, the diff algorithm returns a single insertion at index `-1`. For example, `let changes = ["a"].diff(["a"])` produces
```
(Diff.Diff) $R0 = {
  elements = 1 value {
    [0] = insert {
      insert = (at = -1)
    }
  }
}
```
This could probably be avoided by implementing the "equality pre-processing optimisation" #5, but maybe there is something to check in the core algorithm?

Cheers,
Flo

**Comment by wokalski** [comment]
Oh shit. This is pretty bad. I'm looking into it.

**Comment by wokalski** [comment]
@flovouin Thanks a lot. It's fixed now (not pushed yet). Unfortunately I'm not planning to maintain Swift 2.3 version. If you'd like to do it, all contributions are welcome. I think that it's a waste of time though. 8.2 removes `2.3` version and I expect the vast majority of applications move to `3.0` sooner than later

**Comment by flovouin** [comment]
Thanks! Sure, I completely understand, I'm not expecting you to maintain it. I also agree it's a waste of time. It's just that I have a rather larger codebase to migrate and now is not the time to do it :-)
Feel free to close the issue once the fix is merged.

**Comment by wokalski** [comment]
Fixed in [0.3](https://github.com/wokalski/Diff.swift/releases/tag/0.3)
